src/main/java/me/jellysquid/mods/lithium/common/util/collections/HashSetBackedList.java

```diff
@@ -0,0 +1,187 @@
+package me.jellysquid.mods.lithium.common.util.collections;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+/**
+ * A hacky way to use {@link HashSet} as {@link List} in Sponge's Mixin Environment.
+ *
+ * @param <E> the type of elements maintained by this backed collection
+ *
+ * @author Maity
+ */
+public class HashSetBackedList<E> implements List<E> {
+    private final Collection<E> backedCollection;
+
+    public HashSetBackedList() {
+        this.backedCollection = new HashSet<>();
+    }
+
+    @Override
+    public int size() {
+        return this.backedCollection.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.backedCollection.isEmpty();
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return this.backedCollection.contains(o);
+    }
+
+    @Override
+    public Iterator<E> iterator() {
+        return this.backedCollection.iterator();
+    }
+
+    @Override
+    public void forEach(Consumer<? super E> action) {
+        this.backedCollection.forEach(action);
+    }
+
+    @Override
+    public Object[] toArray() {
+        return this.backedCollection.toArray();
+    }
+
+    @Override
+    public <T> T[] toArray(T[] a) {
+        return this.backedCollection.toArray(a);
+    }
+
+    @Override
+    public boolean add(E e) {
+        return this.backedCollection.add(e);
+    }
+
+    @Override
+    public boolean remove(Object o) {
+        return this.backedCollection.remove(o);
+    }
+
+    @Override
+    public boolean containsAll(Collection<?> c) {
+        return this.backedCollection.containsAll(c);
+    }
+
+    @Override
+    public boolean addAll(Collection<? extends E> c) {
+        return this.backedCollection.addAll(c);
+    }
+
+    @Override
+    public boolean removeAll(Collection<?> c) {
+        return this.backedCollection.removeAll(c);
+    }
+
+    @Override
+    public boolean removeIf(Predicate<? super E> filter) {
+        return this.backedCollection.removeIf(filter);
+    }
+
+    @Override
+    public boolean retainAll(Collection<?> c) {
+        return this.backedCollection.retainAll(c);
+    }
+
+    @Override
+    public void clear() {
+        this.backedCollection.clear();
+    }
+
+    @Override
+    public Spliterator<E> spliterator() {
+        return this.backedCollection.spliterator();
+    }
+
+    @Override
+    public Stream<E> stream() {
+        return this.backedCollection.stream();
+    }
+
+    @Override
+    public Stream<E> parallelStream() {
+        return this.backedCollection.parallelStream();
+    }
+
+    @Override
+    public boolean addAll(int index, Collection<? extends E> c) {
+        return false;
```

Comment:
If any other mod decides to use this method, we shoudn't fail silently if we don't implement it.
I doubt many other mods mess with unloading block entities. Using HashedList might be a good option still.
Otherwise we need to throw UnsupportedOperationExceptions as failing silently can lead to a lot more problems

Reply:
I decided not to use `HashedList` as it uses two collections at once. These are more resources. Of course, this would also speed up the process of unloading block entities, but then the patch was not effective.